sdk/storage/azure-storage-file-datalake/out/azure/storage/filedatalake/_models.pyi

```diff
@@ -0,0 +1,112 @@
+# Stubs for azure.storage.filedatalake._models (Python 3)
+#
+# NOTE: This dynamically typed stub was automatically generated by stubgen.
+
+from ._deserialize import return_headers_and_deserialized_path_list
+from ._generated.models import Path
+from ._shared.models import DictMixin
+from ._shared.response_handlers import process_storage_error
+from azure.core.paging import PageIterator
+from azure.storage.blob import AccessPolicy as BlobAccessPolicy, AccountSasPermissions as BlobAccountSasPermissions, BlobSasPermissions, ContainerSasPermissions, ContentSettings as BlobContentSettings, LeaseProperties as BlobLeaseProperties, ResourceTypes as BlobResourceTypes, UserDelegationKey as BlobUserDelegationKey
+from azure.storage.blob._models import ContainerPropertiesPaged
+from enum import Enum
+from typing import Any, Optional
+
+class FileSystemProperties:
+    name: Any = ...
+    last_modified: Any = ...
+    etag: Any = ...
+    lease: Any = ...
+    public_access: Any = ...
+    has_immutability_policy: Any = ...
+    has_legal_hold: Any = ...
+    metadata: Any = ...
+    def __init__(self) -> None: ...
+
+class FileSystemPropertiesPaged(ContainerPropertiesPaged):
+    def __init__(self, *args: Any, **kwargs: Any) -> None: ...
+
+class DirectoryProperties(DictMixin):
+    name: Any = ...
+    etag: Any = ...
+    deleted: Any = ...
+    metadata: Any = ...
+    lease: Any = ...
+    last_modified: Any = ...
+    creation_time: Any = ...
+    deleted_time: Any = ...
+    remaining_retention_days: Any = ...
+    def __init__(self, **kwargs: Any) -> None: ...
+
+class FileProperties(DictMixin):
+    name: Any = ...
+    etag: Any = ...
+    deleted: Any = ...
+    metadata: Any = ...
+    lease: Any = ...
+    last_modified: Any = ...
+    creation_time: Any = ...
+    size: Any = ...
+    deleted_time: Any = ...
```

Comment:
I wonder if we need both `deleted` and `deleted_time` - as presumably the latter will only be set if the former is True?

Reply:
Possible that `deleted` could be set to True while "pending" deleted before deleted_time is set.
Confirm API service behavior - and update docs accordingly.